nanoFramework.Tools.DebugLibrary.Shared/PortSerial/PortSerial.cs

```diff
@@ -282,13 +333,22 @@ public byte[] ReadBuffer(int bytesToRead)
 
                 try
                 {
-                    int readBytes = Device.Read(buffer, 0, bytesToRead);
+                    // this loop shall fix issues with M1 and some drivers showing up bytes to fast
+                    List<byte> received = new List<byte>();
+                    int readBytes = 0;
+                    while (Device.BytesToRead > 0)

```

Comment:
This condition should also stop reading if all requested bytes (`bytesToRead`) have been read. Otherwise it will start reading data that doesn't belong to this packet, potentially ruining the next one.

Reply:
Fixed.